### PR TITLE
Added a much more explanatory docstring to the module

### LINKED ISSUE #407: Have a quick "what you should look for" as packaging docs
Right now loading up:
https://ops.rtfd.io/

Takes you to a page that has a lot of detail for reference material, but the very first line is just:
```
ops package

The Operator Framework.
```

That feels like a good point to flesh out a little bit as it is a welcome page that we point people to.

I don't think it needs to be terribly verbose, and we can probably take some copy from some of the other documentation and put it there.

It just feels like a miss whenever I tell people about it and load that page, and see that it dumps you straight into a ops.charm.module ActionEvent, which is not where you'd want to start any story.

### PR DESCRIPTION
**Pull request by jnsgruk**
Fixes #407

### CODE REVIEW

**Review by balbirthomas — APPROVED** [review]
LGTM

File: ops/__init__.py [hunk 1]
```diff
@@ -12,9 +12,33 @@
...
+and straightforward way, using standard Python structures to allow for clean,
+maintainable, and reusable code.
+
+A Kubernetes operator is a container that drives lifecycle management,
+configuration, integration and daily actions for an application. Operators
+simplify software management and operations. They capture reusable app domain
+knowledge from experts in a software component that can be shared.
+
+The Charmed Operator Framework extends the "operator pattern" to enable Charmed
+Operators, packaged as and often referred to as “charms”. Charms are not just
```
Comment (benhoyt) [inline ops/__init__.py:27]: Nit: should use normal ASCII `"` quote here (like on the line above), not smart quotes.

**Review by benhoyt — APPROVED** [review]

File: ops/__init__.py (hunk 1 above)
Comment (jameinel) [inline ops/__init__.py:27]:
```suggestion
Operators, packaged as and often referred to as "charms". Charms are not just
```

**Review by jameinel — APPROVED** [review]

File: ops/__init__.py (hunk 1 above)
Comment (jnsgruk) [reply ops/__init__.py]: Agreed, not sure how they slipped in there. Thanks!

Comment (jnsgruk) [reply ops/__init__.py]: Done :)

**Comment by jnsgruk** [comment]
Looks like we're ready to go - thanks both. I don't have special merge powers so whenever works! 🙂

**Pull request merged**